lib/core/CJsonStateRestoreTraverser.cc

```diff
@@ -223,15 +228,58 @@ const std::string& CJsonStateRestoreTraverser::nextValue() const {
 }
 
 bool CJsonStateRestoreTraverser::parseNext(bool remember) {
-    if (m_Reader.HasParseError()) {
-        this->logError();
-        return false;
-    }
+    bool ret{true};
+    SBoostJsonHandler::ETokenType currentTokenType = m_Handler.s_Type;
+    do {
+        if (m_Reader.last_error()) {
+            this->logError();
+            ret = false;
+            break;
+        }
+
+        m_Handler.s_RememberValue = remember;
+
+        char c = '\0';
+        m_ReadStream.get(c);
+        if (c == '\0') {
+            break;
+        }
+
+        // TODO
+//        if (c == '\n') {
+//            m_Handler.s_Type = SBoostJsonHandler::E_TokenWhiteSpace;
+//            continue ;
+//        }
+
+        if (c == '"' && m_Handler.s_Type == SBoostJsonHandler::E_TokenObjectStart) {
+            m_Handler.s_Type = SBoostJsonHandler::E_TokenKey;
+        }
+
+        if (c == ',') {
+            m_Handler.s_Type = SBoostJsonHandler::E_TokenComma;
+        }
+
+        if (c == ':') {
+            m_Handler.s_Type = SBoostJsonHandler::E_TokenColon;
+        }
```

Comment:
It shouldn't be necessary to peek the characters here _except_ checking for `\0` for end-of-document. Doing that would be more conducive to reading blocks of characters rather than one at a time.